### PR TITLE
be tolerant on deployment decode and strict on encode

### PR DESCRIPTION
**Pull request by deads2k**
Allows us to consume legacy v1 serialized deploymentconfigs from annotations, but we only serialize into groupified.

@liggitt resolves rebase problem

### CODE REVIEW

**Comment by deads2k** [comment]
@smarterclayton this is the one you need before you upgrade.

File: pkg/apps/util/scheme.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	// for encoding, we want to be strict on groupified
+	annotationEncodingScheme = runtime.NewScheme()
+	annotationEncoder        runtime.Encoder
+)
+
+func init() {
+	utilruntime.Must(corev1.AddToScheme(annotationDecodingScheme))
+	utilruntime.Must(appsv1.AddToScheme(annotationDecodingScheme))
+	utilruntime.Must(appsv1helpers.AddToScheme(annotationDecodingScheme))
```
Comment (liggitt) [inline pkg/apps/util/scheme.go]: move internal stuff into the TODO block to remove when internal versions go away

File: pkg/apps/util/scheme.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	annotationDecodingScheme = runtime.NewScheme()
+	annotationDecoder        runtime.Decoder
+
+	// for encoding, we want to be strict on groupified
+	annotationEncodingScheme = runtime.NewScheme()
+	annotationEncoder        runtime.Encoder
+)
+
+func init() {
+	utilruntime.Must(corev1.AddToScheme(annotationDecodingScheme))
```
Comment (liggitt) [inline pkg/apps/util/scheme.go]: TODO to move explicit corev1 additions into appsv1.AddToScheme, etc

**Pull request merged**